inst/lib/bs5/scss/_navbar.scss

```diff
@@ -307,11 +307,17 @@
   --#{$prefix}navbar-toggler-border-color: #{$navbar-dark-toggler-border-color};
   --#{$prefix}navbar-toggler-icon-bg: #{escape-svg($navbar-dark-toggler-icon-bg)};
   // scss-docs-end navbar-dark-css-vars
+
+  // patched: toggler icon should follow closest navbar color mode over global mode
+  .navbar-toggler-icon {
+    --#{$prefix}navbar-toggler-icon-bg: #{escape-svg($navbar-dark-toggler-icon-bg)};
+  }
 }
 
 @if $enable-dark-mode {
   @include color-mode(dark) {
-    .navbar-toggler-icon {
+    // patched: toggler follows global theme unless in a light region
+    .navbar:not([data-bs-theme="light"]) .navbar-toggler-icon {
       --#{$prefix}navbar-toggler-icon-bg: #{escape-svg($navbar-dark-toggler-icon-bg)};
     }
```

Comment:
Is this gonna work sensibly for scenarios where we don't have control over the navbar markup (e.g., pkgdown, rmarkdown, flexdashboard, etc.)?

Reply:
I can look into this and will add anything that needs to be addressed to https://github.com/posit-dev/brand-yml/issues/52. Bringing pkgdown and bslib back into alignment is part of that plan.

Reply:
This whole PR does interact with rmarkdown websites and the approach I'd like to take is to make `.navbar-default` and `.navbar-inverse` be no-op classes for BS 5+. We could also take them out of the markup when we know that we're in BS 5+. That's going to require another pass at this PR, though.